src/dtemplate.d

```diff
@@ -4105,6 +4109,44 @@ extern (C++) MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplatePara
                 result = match2;
                 return;
             }
+
+            /* Deduce common type
```

Comment:
Not all. Because `typeMerge` considers alias this of the operands, but in here it's unneeded (it's handled in deduceFunctionTemplateMatch, around line 1540 in dtemplate.d).